web/src/pages/OGTagPreviewPage/OGTagPreviewPage.tsx

```diff
@@ -1,180 +1,112 @@
 import { useState } from 'react'
 
-import { useLazyQuery } from '@apollo/client'
-import {
-  Title,
-  Grid,
-  Col,
-  Card,
-  Text,
-  Flex,
-  Button,
-  TextInput,
-  Tab,
-  TabGroup,
-  TabList,
-  TabPanel,
-  TabPanels,
-  Callout,
-} from '@tremor/react'
+import { Title, Text, Callout } from '@tremor/react'
 
-import { Metadata } from '@redwoodjs/web'
+import { Metadata, useQuery } from '@redwoodjs/web'
 
-import {
-  CodeBracketIcon,
-  ErrorIcon,
-  EyeIcon,
-  LinkIcon,
-  RectangleStackIcon,
-  RefreshIcon,
-} from 'src/icons/Icons'
+import { ErrorIcon } from 'src/icons/Icons'
 
-const OG_TAG_PREVIEW_QUERY = gql`
-  query OGTagPreview($url: String!, $customUserAgent: String) {
-    ogTagPreview(url: $url, customUserAgent: $customUserAgent) {
+import type {
+  OGTagPreviewProviderAudit,
+  OGTagPreviewResponse,
+} from '../../../types/graphql'
+
+import { PreviewFetcher } from './PreviewFetcher'
+import { PreviewTabs } from './PreviewTabs'
+
+const PROJECT_CONFIG_QUERY = gql`
+  query UserProjectConfigQuery {
+    config: userProjectConfig {
       id
-      userAgent
-      error
-      result
+      web {
+        id
+        host
+        port
+        apiUrl
+      }
+      ssr {
+        id
+        enabled {
+          status
+          message
+        }
+      }
     }
   }
 `
 
-function isValidHttpUrl(possibleURL: string) {
-  let url
-  try {
-    url = new URL(possibleURL)
-  } catch (_error) {
-    return false
-  }
-  return url.protocol === 'http:' || url.protocol === 'https:'
-}
-
 const OgTagPreviewPage = () => {
-  const [url, setUrl] = useState('http://localhost:8910/')
+  const [url, setUrl] = useState<string>('')
+  const [showPreviewer, setShowPreviewer] = useState<boolean | null>(null)
   const [customUserAgent, setCustomUserAgent] = useState<string | null>(null)
+  const [error, setError] = useState<Error | null>(null)
+  const [result, setResult] = useState<OGTagPreviewResponse['result'] | null>(
+    null
+  )
+  const [audits, setAudits] = useState<OGTagPreviewProviderAudit[] | null>(null)
 
-  const [getOGTagPreview, ogTagPreviewQuery] =
-    useLazyQuery(OG_TAG_PREVIEW_QUERY)
+  useQuery(PROJECT_CONFIG_QUERY, {
```

Comment:
How come you're not using a Cell for this?

Reply:
Because I just wanted to set a flag to show the disabled message section or render the page as per usual. There is no concept of empty state of need for a spinner loading state. I'd then have to wrap everything in a cell and that didn't seem worth it.

Reply:
So how do you handle loading state now?

Reply:
To me this looked like a typical example of the "CellPage" people have been asking about for a few years now from time to time. I.e. the page would be empty pretty much, just render a Cell that has all the actual page content inside

Reply:
I don't have a loading state - the success and error are done in the OnCompleted and onError handers. Apps don't need to use Cells - we know of very large production apps that do not use that pattern. I would use query and not a cell if I was fetching the nav emu (and did in a preview implementation) as I didn't want to show a loading indication when making menus.

Reply:
Also, unless I used the onCompleted and onError handlers, when I set/modifed the variable like showDisplayScrion etc and that was used in a useEffect, React would complain about multiple renders because the data and loading return values from the useQuery somehow caused multiple renders of the page. This was the cleanest implementation to simply "get the user config" and present the page according to its info.

Reply:
> I would use query and not a cell if I was fetching the nav emu (and did in a preview implementation) as I didn't want to show a loading indication when making menus.

Interesting!

I use a cell for loading the menu on https://acm.se. It actually does take long enough to load it so you can see the loading state, even on pretty fast broadband. In a mobile connection it takes even longer. So to me a nav menu that needs to do a gql request is a perfect use case for a cell 🙂 

Not saying one way or the other is right or wrong. As always, "it depends" 😄

Reply:
Hmm now that I think on it given the refactoring I did to make the previewed component it probably could be a cell where …. Well the main query to fetch the config would still be a success … y’know I’ll re think this a bit. A good reason to make in a cell later is if we make some preview images and that’s an RSC component and then a cell fetch pattern would be good to test. As you said — always options.